dora/src/stdlib.rs

```diff
@@ -249,18 +249,28 @@ pub extern "C" fn argv(ind: i32) -> Ref<Str> {
     panic!("argument does not exist");
 }
 
-pub extern "C" fn str_parse_int(val: Ref<Str>) -> i32 {
+pub extern "C" fn str_to_int_or_throw(val: Ref<Str>) -> i32 {
     let slice = val.content();
     let val = str::from_utf8(slice).unwrap();
 
-    val.parse::<i32>().unwrap_or(0)
+    let result = val.parse::<i32>();
+    if result.is_ok() {
+        return result.unwrap()
+    }
+    throw_native(true);
+    0
```

Comment:
Can't use `unreachable!()`, this is actually reached!?

Reply:
Yes, throwing an exception from native code just sets a flag. When returning back to Dora code, there is assembly code that checks this flag and starts the exception handling process if it is set.